test/unittest/dds/subscriber/SubscriberTests.cpp

```diff
@@ -264,15 +264,235 @@ TEST(SubscriberTests, ChangeDefaultDataReaderQos)
+    qos.durability_service().max_samples = 5;
+    qos.durability_service().max_instances = 20;
+    qos.durability_service().max_samples_per_instance = 30;
+    // .reliable_reader_qos
+    qos.reliable_reader_qos().times.initialAcknackDelay.seconds = 34;
+    qos.reliable_reader_qos().times.initialAcknackDelay.nanosec = 32;
+    qos.reliable_reader_qos().times.heartbeatResponseDelay.seconds = 432;
+    qos.reliable_reader_qos().times.heartbeatResponseDelay.nanosec = 43;
+    qos.reliable_reader_qos().disable_positive_ACKs.enabled = true;
+    qos.reliable_reader_qos().disable_positive_ACKs.duration.seconds = 13;
+    qos.reliable_reader_qos().disable_positive_ACKs.duration.nanosec = 320;
+    // .type_consistency
+    qos.type_consistency().representation.m_value.push_back(XML_DATA_REPRESENTATION);
+    qos.type_consistency().representation.m_value.push_back(XCDR_DATA_REPRESENTATION);
+    qos.type_consistency().type_consistency.m_ignore_sequence_bounds = false;
+    qos.type_consistency().type_consistency.m_ignore_string_bounds = false;
+    qos.type_consistency().type_consistency.m_ignore_member_names = true;
+    qos.type_consistency().type_consistency.m_prevent_type_widening = true;
+    qos.type_consistency().type_consistency.m_force_type_validation = true;
+    // .expects_inline_qos
+    qos.expects_inline_qos(true);
+    // .properties
+    eprosima::fastrtps::rtps::Property property;
+    property.name("Property1");
+    property.value("Value1");
+    qos.properties().properties().push_back(property);
+    property.name("Property2");
+    property.value("Value2");
+    qos.properties().properties().push_back(property);
+
+    // .endpoint
+    qos.endpoint().user_defined_id = 1;
+    qos.endpoint().entity_id = 2;
+    qos.endpoint().history_memory_policy = eprosima::fastrtps::rtps::PREALLOCATED_WITH_REALLOC_MEMORY_MODE;
+
+    // .reader_resource_limits
+    qos.reader_resource_limits().matched_publisher_allocation.initial = 30;
+    qos.reader_resource_limits().matched_publisher_allocation.maximum = 300;
+    qos.reader_resource_limits().matched_publisher_allocation.increment = 4;
+    qos.reader_resource_limits().sample_infos_allocation.initial = 40;
+    qos.reader_resource_limits().sample_infos_allocation.maximum = 400;
+    qos.reader_resource_limits().sample_infos_allocation.increment = 5;
+    qos.reader_resource_limits().outstanding_reads_allocation.initial = 50;
+    qos.reader_resource_limits().outstanding_reads_allocation.maximum = 500;
+    qos.reader_resource_limits().outstanding_reads_allocation.increment = 6;
+    qos.reader_resource_limits().max_samples_per_read = 33;
+
+    // .data_sharing
+    qos.data_sharing().on("/");
 
     ASSERT_TRUE(subscriber->set_default_datareader_qos(qos) == ReturnCode_t::RETCODE_OK);
 
     DataReaderQos wqos;
     subscriber->get_default_datareader_qos(wqos);
 
+    // .durability
+    ASSERT_EQ(eprosima::fastdds::dds::TRANSIENT_DURABILITY_QOS, wqos.durability().kind);
```

Comment:
I would change all the `ASSERT_XXX` into `EXPECT_XXX`

Reply:
Done